### PR TITLE
Update DSC configuration to use OOP API

### PR DESCRIPTION
**Pull request by AmelBawa-msft**
## Summary of the pull request
- Latest `Microsoft.Management.Configuration` nuget allows applying DSC configuration in out of proc context.
- Remove powershell dependency as it's no longer needed for OOP proc configuration execution
- Updated code to use the updated Configuration API
- Ensure configuration is unstubbed during initialization

## PR checklist
- [ ] Closes #xxx
- [ ] Tests added/passed
- [ ] Documentation updated

### CODE REVIEW

**Review by EricJohnson327 — APPROVED** [review]

File: tools/SetupFlow/DevHome.SetupFlow.Common/TelemetryEvents/ConfigurationUnitResultEvent.cs [hunk 1]
```diff
@@ -15,7 +15,7 @@ internal sealed class ConfigurationUnitResultEvent : EventBase
 {
     private readonly ApplyConfigurationUnitResult _unitResult;
 
-    public string UnitName => _unitResult.Unit.UnitName;
+    public string UnitName => _unitResult.Unit.Type;
```
Comment (EricJohnson327) [inline tools/SetupFlow/DevHome.SetupFlow.Common/TelemetryEvents/ConfigurationUnitResultEvent.cs:18]: NIT: Should this variable still be called UnitName?

Comment (AmelBawa-msft) [reply tools/SetupFlow/DevHome.SetupFlow.Common/TelemetryEvents/ConfigurationUnitResultEvent.cs:18]: `UnitName` was renamed to `Type` in this [commit](https://github.com/microsoft/winget-cli/commit/721c81ff47c12c454c4e8d973701b7899f2f22ef#diff-079d03a1599653ae5a9de8e2159c9573b169abb50f75012628f2ea3fa6f2adb0L49-R62)

Comment (AmelBawa-msft) [reply tools/SetupFlow/DevHome.SetupFlow.Common/TelemetryEvents/ConfigurationUnitResultEvent.cs]: Renamed `UnitName` to `Type` to align with the changes. Thank you!

File: tools/SetupFlow/DevHome.SetupFlow.Common/DevHome.SetupFlow.Common.csproj [hunk 2]
```diff
@@ -49,6 +50,9 @@
       <GeneratePathProperty>true</GeneratePathProperty>
       <IncludeAssets>none</IncludeAssets>
     </PackageReference>
+    <PackageReference Include="Microsoft.WindowsPackageManager.Configuration.OutOfProc" Version="1.7.10091-preview">
```
Comment (krschau) [inline tools/SetupFlow/DevHome.SetupFlow.Common/DevHome.SetupFlow.Common.csproj:53]: Is it ok to be using preview packages?

File: tools/SetupFlow/DevHome.SetupFlow.UnitTest/BaseSetupFlowTest.cs [hunk 3]
```diff
@@ -1,87 +1,88 @@
 ﻿// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
-
-using DevHome.Common.Extensions;
```
Comment (krschau) [inline tools/SetupFlow/DevHome.SetupFlow.UnitTest/BaseSetupFlowTest.cs]: Nit: you can remove some of the unused usings in this file

**Review by krschau — APPROVED** [review]

File: tools/SetupFlow/DevHome.SetupFlow.Common/DevHome.SetupFlow.Common.csproj (hunk 2 above)
Comment (AmelBawa-msft) [reply tools/SetupFlow/DevHome.SetupFlow.Common/DevHome.SetupFlow.Common.csproj:53]: Yes this is ok, we've requested this package 🙂

**Pull request merged**